src/providers/drafts.ts

```diff
@@ -0,0 +1,123 @@
+  removeDrafts: (draftIds: string[]) => void
+  removeAllDrafts: (channelId: string) => void
+  markAllDraftsAsSeenForChannel: (channelId: string) => void
+}
+
+export const useDraftStore = createStore<DraftStoreStateV1, DraftStoreActions>(
+  {
+    state: {
+      allDrafts: [], // includes drafts for different channels
+    },
+    actionsFactory: (set) => ({
+      addDraft: (draft, explicitId) => {
+        const id = explicitId ?? createId()
+        const updatedAt = new Date().toISOString()
+        const newDraft: Draft = { ...draft, updatedAt, id, seen: false }
+        set((draftState) => {
+          draftState.allDrafts.unshift(newDraft)
+        })
+        return newDraft
+      },
+      updateDraft: (draftId, draftProps) => {
+        const updatedAt = new Date().toISOString()
+        set((draftState) => {
+          const idx = draftState.allDrafts.findIndex((d) => d.id === draftId)
+          if (idx >= 0) {
+            draftState.allDrafts[idx] = { ...draftState.allDrafts[idx], ...draftProps, updatedAt }
+          }
+        })
+      },
+      removeDrafts: (draftIds) => {
+        set((draftState) => {
+          draftState.allDrafts = draftState.allDrafts.filter((draft) => !draftIds.includes(draft.id))
+        })
+      },
+      removeAllDrafts: (channelId) => {
+        set((draftState) => {
+          draftState.allDrafts = draftState.allDrafts.filter((draft) => draft.channelId !== channelId)
+        })
+      },
+      markAllDraftsAsSeenForChannel: (channelId) => {
+        set((draftState) => {
+          draftState.allDrafts = draftState.allDrafts.map((draft) => ({
+            ...draft,
+            seen: draft.channelId === channelId ? true : draft.seen,
+          }))
+        })
+      },
+    }),
+  },
+  {
+    persist: {
+      key: 'drafts',
+      whitelist: ['allDrafts'],
+      version: 1,
+      migrate: (oldState, oldVersion, storageValue) => {
+        // migrate store before zustand was added
+        if (oldVersion === undefined) {
+          const unseenDrafts = readFromLocalStorage<
+            Array<{
+              draftId: string
+              channelId: string
+            }>
+          >('unseenDrafts')
+          const drafts = [...(storageValue as Array<Draft>)].map((draft) => {
+            unseenDrafts?.find((unseen) => unseen.draftId === draft.id)
+              ? { ...draft, seen: false }
+              : { ...draft, seen: true }
+          })
+          return {
+            allDrafts: drafts,
+          }
+        }
+      },
+    },
+  }
+)
+
+export const getDraft = (id: string) => (store: CommonStore<DraftStoreStateV1, DraftStoreActions>) =>
+  store.allDrafts.find((draft) => draft.id === id)
+export const getDraftsForChannel = (channelId: string) => (store: CommonStore<DraftStoreStateV1, DraftStoreActions>) =>
+  store.allDrafts.filter((d) => d.channelId === channelId)
+export const getUnseenDraftsForChannel = (channelId: string) => (
```

Comment:
I like this pattern but let's improve upon the naming here. I propose:
- `singleDraftSelector`
- `channelDraftsSelector`
- `chanelUnseenDraftsSelector`